### PR TITLE
Add LSTM model

### PR DESCRIPTION
**Pull request by jacobbieker**
Adds LSTM model for forecasting ELUC, same was as the XGBoost one, and with same data processing. The LSTM takes the last decade of data and predicts for the next year.

Performance (143 is UK, 29 is Brazil):
```
Test  143 MSE: 5.043389550480362e-05
Test 143 MAE: 0.0003478719354466271
Test  29 MSE: 0.0015637133995812633
Test 29 MAE: 0.0016567066047134232
Test  [143, 29] MSE: 0.0007422277407239995
Test [143, 29] MAE: 0.0009677064777143655
```

### CODE REVIEW

File: model/lstm.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        x = torch.relu(x)
+        x = torch.squeeze(x)
+        x = self.average(x)
+        return x
+
+
+def train_country_lstm(country_code):
+    test_da = dataset.where(country_mask == country_code, drop=True).where(dataset.time > 2007, drop=True).load()
+    train_da = dataset.where(country_mask == country_code, drop=True).where(dataset.time <= 2007, drop=True).load() # 143 is the code for the UK
+    model = LSTMModel().cuda()
```
Comment (ofrancon) [inline model/lstm.py:63]: Does it mean we need a Nvidia graphic card to train it?

Comment (ofrancon) [reply model/lstm.py:63]: And at inference time?

**Review by ofrancon — APPROVED** [review]
lgtm. Much better mse/mae than XGBoost!
Will be interesting to check what ELUC it predicts when there is no land usage change in that year. (expecting 0 or close to 0).

File: model/lstm.py (hunk 1 above)
Comment (jacobbieker) [reply model/lstm.py]: I've changed it to select whatever device is available, GPU or CPU, to train on and for the test inference. It is much faster on the GPU, but not terribly slow on CPU either.

**Pull request merged**